internal/controllers/topology/names/namegenerator.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package names provides functions to generate object names.
+package names
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/pkg/errors"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apiserver/pkg/storage/names"
+)
+
+const (
+	maxNameLength          = 63
+	randomLength           = 5
+	maxGeneratedNameLength = maxNameLength - randomLength
+)
+
+type simpleNameGenerator struct {
+	base string
+}
+
+func (s *simpleNameGenerator) GenerateName() (string, error) {
+	return names.SimpleNameGenerator.GenerateName(s.base), nil
```

Comment:
Not sure but maybe we should just inline the GenerateName func of the SimpleNameGenerator

This way this uses the same constants as the code below